gapis/replay/wait_for_profiler.go

```diff
@@ -0,0 +1,47 @@
+// Copyright (C) 2019 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package replay
+
+import (
+	"context"
+
+	"github.com/google/gapid/gapis/api"
+	"github.com/google/gapid/gapis/api/transform"
+	"github.com/google/gapid/gapis/replay/builder"
+)
+
+type WaitForProfiler struct{}
+
+func (t *WaitForProfiler) AddResult(r Result) {
+}
+
+func (t *WaitForProfiler) Transform(ctx context.Context, id api.CmdID, cmd api.Cmd, out transform.Writer) {
+	out.MutateAndWrite(ctx, id, cmd)
+}
+
+func (t *WaitForProfiler) Flush(ctx context.Context, out transform.Writer) {
+	t.AddWaitInstruction(ctx, out)
```

Comment:
This will only add wait instruction at the end, which is not really useful. Can you extend this to insert the wait instruction just after a command with a given ID? This would match with some of the other transforms we have (such as the read framebuffer one) and already allow us to use it.